apps/web/src/app/[orgShortcode]/settings/user/security/_components/RecoveryEmailSection.tsx

```diff
@@ -0,0 +1,223 @@
+'use client';
+
+import {
+  InputOTP,
+  InputOTPGroup,
+  InputOTPSlot
+} from '@/src/components/shadcn-ui/input-otp';
+import {
+  DisableRecoveryEmailModal,
+  RecoveryEmailModal
+} from './recovery-email-modals';
+import { Button } from '@/src/components/shadcn-ui/button';
+import { CheckCircle, Clock } from '@phosphor-icons/react';
+import { REGEXP_ONLY_DIGITS_AND_CHARS } from 'input-otp';
+import { platform } from '@/src/lib/trpc';
+import { format } from 'date-fns';
+import { useState } from 'react';
+import { toast } from 'sonner';
+
+interface RecoveryEmailSectionProps {
+  ensureElevated: (fn: () => Promise<void> | void) => void;
+}
```

Comment:
Just import this function form the other file, don't pass it as props, that causes unnecessary re-renders

Reply:
good point wasn't happy with about passing that